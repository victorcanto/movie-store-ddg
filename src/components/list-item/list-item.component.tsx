import * as React from 'react';
import {
	Box,
	Grid,
	IconButton,
	ListItem as MuiListItem,
	ListItemText,
	Tooltip,
	Typography,
} from '@mui/material';
import DeleteIcon from '@mui/icons-material/Delete';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
import { CartItem, FavoriteItem } from '../../shared';
import { selectIsFavoritesOpen, useAppSelector } from '../../store';
import { Price } from '../price/price.component';

export interface ListItemProps {
	item: CartItem | FavoriteItem;
	removeItem: () => void;
	addItemToCart?: () => void;
}

export function ListItem({ item, removeItem, addItemToCart }: ListItemProps) {
	const isFavoritesOpen = useAppSelector(selectIsFavoritesOpen);

	return (
		<Grid data-testid='list-item' container spacing={3} alignItems='center'>
			<Grid item xs={2}>
				<img
					width='50px'
					src={`https://image.tmdb.org/t/p/w500/${item?.poster_path}`}
					alt={item?.title}
				/>
			</Grid>
			<Grid item xs={4}>
				<Typography>{item.title}</Typography>
			</Grid>
			<Grid item xs={2}>
				<Typography>{1}</Typography>
			</Grid>
			<Grid item xs={2}>
				<Price value={item.price} />
			</Grid>
			<Grid item xs={2}>
				<Box>
					{isFavoritesOpen ? (
						<IconButton
							onClick={addItemToCart}
							edge='end'
							aria-label='add to cart'
						>
							<ShoppingCartIcon />
						</IconButton>
					) : null}
					<Tooltip title='Remover do carrinho'>
						<IconButton
							onClick={removeItem}
							edge='end'
							aria-label='delete'
						>
							<DeleteIcon />
						</IconButton>
					</Tooltip>
				</Box>
			</Grid>
		</Grid>
	);
}
