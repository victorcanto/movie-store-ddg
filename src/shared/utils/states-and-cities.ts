import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
