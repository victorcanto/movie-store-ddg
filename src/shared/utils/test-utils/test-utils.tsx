import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
