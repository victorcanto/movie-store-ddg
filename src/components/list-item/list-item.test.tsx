import React from 'react';
import { render } from '../../shared/utils/test-utils';
import { ListItem, ListItemProps } from './list-item.component';

const propsMock: ListItemProps = {
	item: {
		id: 1,
		poster_path: 'test',
		price: 79.99,
		title: 'test',
		quantity: 1,
	},
	removeItem: jest.fn(),
};

test('snapshot of the base state', () => {
	const { asFragment } = render(<ListItem {...propsMock} />);
	expect(asFragment()).toMatchSnapshot();
});

test('should have in the document', () => {
	const { getByTestId } = render(<ListItem {...propsMock} />);
	expect(getByTestId('list-item')).toBeInTheDocument();
});
