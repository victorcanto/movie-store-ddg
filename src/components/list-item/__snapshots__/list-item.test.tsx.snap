// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-18ksmt-MuiGrid-root"
    data-testid="list-item"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
    >
      <img
        alt="test"
        src="https://image.tmdb.org/t/p/w500/test"
        width="50px"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        test
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        1
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
    >
      <span
        data-testid="price"
      >
        R$79.99
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <button
          aria-label="delete"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DeleteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
